src/dfi/poolpairs.cpp

```diff
@@ -241,6 +241,50 @@ auto InitPoolVars(CPoolPairView &view, PoolHeightKey poolKey, uint32_t end) {
     return std::make_tuple(std::move(value), std::move(it), height);
 }
 
+static auto GetRewardPerShares(const CPoolPairView &view, const TotalRewardPerShareKey &key) {
+    return std::make_tuple(view.GetTotalRewardPerShare(key),
+                           view.GetTotalLoanRewardPerShare(key),
+                           view.GetTotalCommissionPerShare(key),
+                           view.GetTotalCustomRewardPerShare(key));
+}
+
+void CPoolPairView::CalculateStaticPoolRewards(std::function<CAmount()> onLiquidity,
+                                               std::function<void(RewardType, CTokenAmount, uint32_t)> onReward,
+                                               const uint32_t poolID,
+                                               const uint32_t beginHeight,
+                                               const uint32_t endHeight) {
+    if (beginHeight >= endHeight) {
+        return;
+    }
+
+    // Get start and end reward per share
+    TotalRewardPerShareKey key{beginHeight, poolID};
+    auto [startCoinbase, startLoan, startCommission, startCustom] = GetRewardPerShares(*this, key);
+    key.height = endHeight - 1;
+    auto [endCoinbase, endLoan, endCommission, endCustom] = GetRewardPerShares(*this, key);
+
+    // Get owner's liquidity
+    const auto liquidity = onLiquidity();
+
+    auto calcReward = [&](RewardType type, const arith_uint256 &start, const arith_uint256 &end, const uint32_t id) {
+        if (const auto rewardPerShare = end - start; rewardPerShare > 0) {
+            // Calculate reward
+            const auto reward = (liquidity * rewardPerShare / HIGH_PRECISION_SCALER).GetLow64();
+            // Pay reward to the owner
+            onReward(type, {DCT_ID{id}, static_cast<CAmount>(reward)}, endHeight);
+        }
+    };
+
+    calcReward(RewardType::Coinbase, startCoinbase, endCoinbase, {});
+    calcReward(RewardType::LoanTokenDEXReward, startLoan, endLoan, {});
```

Comment:
better to explicitly give DFI_ID(0) than {} ? IMHO such defaults make it harder to spot errors.